api/lib/domain/services/verify-certificate-code-service.js

```diff
@@ -0,0 +1,22 @@
+const _ = require('lodash');
+const certificationCourseRepository = require('../../../lib/infrastructure/repositories/certification-course-repository');
+
+const availableCharacters = '123456789ABCDEFGHJKLMNPQRSTUVWXZ'.split('');
```

Comment:
le 0 et le O ne sont pas mis, il me semblait que dans les code, seul le 0 pouvait être mis

Reply:
En effet, vu que le code a pour but d'être partagé (numériquement ou recopié à la main) on s'était dit que le 0 et O pouvaient être trop facilement confondu.
De la même manière on a écarté le "i" majuscule (pouvant être confondu avec un 'L' minuscule).

Reply:
ma remarque porte sur "pourquoi on enlève les 2 ?", si on compare avec l'ambiguïté "I" et "L", le "I" a été enlevé mais pas le "L". Ici on pourrait faire pareil et garder le zéro (il me semble que c'est la norme ping @laura-bergoens, qui m'en avait parlé)

Reply:
:grimacing: me souviens plus

Reply:
En fait je me souviens d'une discussion aaaancienne avec @jonathanperret (au doigt mouillé c'était vers octobre 2019) qui me rappelait qu'il existe tout un tas d'études autour du sujet de la génération de code alphanumérique _human-friendly_, et qu'il fallait pas réinventer la roue et s'en inspirer.

J'ai donc parcouru les internets et je suis tombée sur des ressources / infos intéressantes, notamment :
- Il semblerait que Microsoft, pour ses product-keys (qui fonctionnent bien depuis des dizaines d'années) ont décidé de prendre les caractères suivants : 2346789BCDFGHJKMPQRTVWXY  et donc d'ignorer les suivants : 015AEILNOSUZ (source de 2ème main ici -> https://security.stackexchange.com/questions/40226/are-there-any-guidelines-for-passwords-that-need-to-be-readout-loud). Note: on remarquera que les voyelles ont été retirées sur la recommandation Microsoft. La raison a cela c'est pour tout simplement éviter de générer des codes contenant des mots compréhensibles, et surtout potientiellement offensant (NEGRE123, SALGROS789 ou COUILLE159 ne sont pas souhaitables).
- L'UX peut **largement** contribué à la bonne diffusion / compréhension du code. Le code est-il aussi destiné à être imprimé ? ou seulement lu sur l'écran ? Il semblerait que les polices de type Sans Sérif soit mieux pour les écrans, mais les Roman-style Serif mieux pour l'impression. Ne pas utiliser d'italique. Ne pas souligner. Avoir un bon spacing entre les caractères à l'affichage. (article sur la font ledgibility -> https://www.fonts.com/content/learning/fontology/level-4/fine-typography/legibility)
-

Reply:
- On peut le présenter par chunk ? (par groupe 3 ou 4 caractères)

Reply:
Bon, du coup ça limite un peu le nombre de combinaisons mais je pense qu'il y'a un trade-off à trancher ici que j'ai vu appeler `non-ambiguity over brevity`. C'est plus frustrant pour un utilisateur de taper un code et qu'il soit "mal saisi", que de frapper quelques caractères de plus

Reply:
Et enfin, voici apparement `THE FONT` pour la `character-recognition` --> https://en.wikipedia.org/wiki/OCR-A

Reply:
Mes multiples clics et parcours d'internet ont débutés avec ce lien -> https://github.com/dwyl/cid/issues/2